Test/CoreSDK.Test/Net40/TaskExTests.cs

```diff
@@ -0,0 +1,173 @@
+namespace Microsoft.ApplicationInsights
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class TaskExTests
+    {
+
+        [TestMethod]
+        public void RethrowIfFailedDoesntThrowIfNoExceptionOccured()
+        {
+            var task = Task.Factory.StartNew(() => { });
```

Comment:
nitpick: I know this is a hotly debated topic online but I respectfully ask that you use concrete types instead of var as it increases readability of the code (especially outside of Visual Studio where there is no IntelliSense).